ci-operator/step-registry/idp/htpasswd/idp-htpasswd-commands.sh

```diff
@@ -3,6 +3,49 @@
 set -e
 set -u
 set -o pipefail
+set -x
+
+function check_if_hypershift_env () {
+    if [ -f "${SHARED_DIR}/nested_kubeconfig" ]; then
+        IS_HYPERSHIFT_ENV="yes"
+    else
+        IS_HYPERSHIFT_ENV="no"
+        return 0
+    fi
+    MC_KUBECONFIG_FILE="${SHARED_DIR}/hs-mc.kubeconfig"
+    if [ -f "${MC_KUBECONFIG_FILE}" ]; then
+        export KUBECONFIG="${SHARED_DIR}/hs-mc.kubeconfig"
+        _jsonpath="{.items[?(@.metadata.name==\"$(cat ${SHARED_DIR}/cluster-name)\")].metadata.namespace}"
+        HYPERSHIFT_NAMESPACE=$(oc get hostedclusters -A -ojsonpath="$_jsonpath")
+    elif [ -f "${SHARED_DIR}/mgmt_kubeconfig" ]; then
+        export KUBECONFIG="${SHARED_DIR}/mgmt_kubeconfig"
+    else
+        export KUBECONFIG="${SHARED_DIR}/kubeconfig"
+    fi
+    
+    count=$(oc get hostedclusters --no-headers --ignore-not-found -n "$HYPERSHIFT_NAMESPACE" | wc -l)
+    echo "hostedcluster count: $count"
+    if [ "$count" -lt 1 ]  ; then
+        echo "namespace clusters don't have hostedcluster"
+        exit 1
+    fi
+    # Limitation: we always & only select the first hostedcluster to add idp-htpasswd. "
+    cluster_name=$(oc get hostedclusters -n "$HYPERSHIFT_NAMESPACE" -o jsonpath='{.items[0].metadata.name}')
+}
+
+function set_common_variables () {
+    if [ "$IS_HYPERSHIFT_ENV" == "yes" ]; then
+        MIDDLE_NAMESPACE="$HYPERSHIFT_NAMESPACE"
+        TARGET_RESOURCE="hostedcluster/$cluster_name -n $MIDDLE_NAMESPACE"
+        OAUTH_NAMESPACE="clusters-$cluster_name"
```

Comment:
For the following Hypershift CI, the hostedcluster namespace is not clusters, but local-cluster.

Reply:
So I think OAUTH_NAMESPACE="clusters-$cluster_name" should change to
```
OAUTH_NAMESPACE="$MIDDLE_NAMESPACE-$cluster_name"
```